infrastructure/azure.tf

```diff
@@ -533,6 +544,26 @@ resource "null_resource" "azurerm_app_service_portal_git" {
   }
 }
 
+locals {
+  application_outbound_ips = "${azurerm_shared_address_space_cidr},${azurerm_function_app.azurerm_function_app.outbound_ip_addresses},${azurerm_app_service.azurerm_app_service_portal.outbound_ip_addresses}"
+}
+
+resource "null_resource" "azurerm_cosmosdb_ip_range_filter" {
+  triggers = {
+    cosmosdb_id = "${azurerm_cosmosdb_account.azurerm_cosmosdb.id}"
+    application_outbound_ips = "${local.application_outbound_ips}"
+
+    # increment the following value when changing the provisioner script to
+    # trigger the re-execution of the script
+    # TODO: consider using the hash of the script content instead
+    provisioner_version = "1"
+  }
+
+  provisioner "local-exec" {
```

Comment:
unfortunately there's a cyclic dependency:

* the function app resource depends on the cosmosdb resource (for the connection strings)
* the cosmosdb resource depends on the function app (for the outbound IPs)

Reply:
not sure if it's an option, but it could be fixed by setting the cosmosdb primary key through a variable instead of autogenerating it- need to check if possible

Reply:
not sure if it's an option, but it could be fixed by setting the cosmosdb primary key through a variable instead of autogenerating it- need to check if possible